build-system/tasks/e2e/validate-tests.js

```diff
@@ -0,0 +1,239 @@
+      : 95) / 100
+  );
+}
+
+/**
+ * Get the modified test files which require validation.
+ * This can be manually overridden with the command line argument `files` to
+ * validate arbitrary files.
+ * @return {string[]}
+ */
+function getTestsFilesToValidate() {
+  if (argv.files) {
+    return getFilesFromArgv();
+  }
+  const changedFiles = gitDiffNameOnlyMain();
+  const testFilesRequiringValidation = getDefaultMochaFiles().filter((file) =>
+    changedFiles.includes(file)
+  );
+
+  return testFilesRequiringValidation;
+}
+
+/**
+ * @param {Record<string, TestResultDef>[]} results
+ * @return {Record<string, number>}
+ */
+function computeAverageTestRuntimes(results) {
+  // Computing runtimes by test.
+  const totalTestRuntimes = results.reduce((testTimes, result) => {
+    Object.entries(result).forEach(([testName, testInfo]) => {
+      if (!testTimes[testName]) {
+        testTimes[testName] = 0;
+      }
+      testTimes[testName] += testInfo.endTime - testInfo.startTime;
+
+      return testTimes;
+    }, {});
+    return testTimes;
+  }, /** @type {Record<string, number>} */ ({}));
+
+  return Object.entries(totalTestRuntimes).reduce(
+    (averageRuntimes, [testName, totalRuntime]) => {
+      averageRuntimes[testName] = parseFloat(
+        (totalRuntime / results.length).toFixed(2)
+      );
+      return averageRuntimes;
+    },
+    {}
+  );
+}
+
+/**
+ * @param {Record<string, TestResultDef>[]} results
+ * @return {Record<string, number>}
+ */
+function computeTestFlakiness(results) {
+  const testPassCounts = results.reduce((nameToPassCount, result) => {
+    Object.entries(result).forEach(([testName, testInfo]) => {
+      if (!nameToPassCount[testName]) {
+        nameToPassCount[testName] = 0;
+      }
+      nameToPassCount[testName] += testInfo.passed ? 1 : 0;
+    });
+
+    return nameToPassCount;
+  }, /** @type {Record<string, number>} */ ({}));
+  return Object.entries(testPassCounts).reduce(
+    (testPassPercentages, [testName, passCount]) => {
+      testPassPercentages[testName] = (passCount / results.length).toFixed(4);
+      return testPassPercentages;
+    },
+    {}
+  );
+}
+
+/**
+ * Runs the modified tests the configured number of times and returns their speeds and success rates.
+ * @return {Promise<Record<string, TestResultDef>[]|void>}
+ */
+async function runValidatorIterations() {
+  // specify tests to run
+  const testFilesRequiringValidation = getTestsFilesToValidate();
+
+  if (!testFilesRequiringValidation.length) {
+    log(green('No Tests Require Validation'));
+    return;
+  }
+  log(
+    green('Test Files Requiring Validation:\n'),
+    testFilesRequiringValidation.join('\n')
+  );
+
+  /** @type {Record<string, TestResultDef>[]} */
+  const results = [];
+  await /** @type {Promise<void>} */ (
+    new Promise((resolve) => {
+      const iterations = getTestValidatorIterations_();
+      log(green(`Creating ${iterations} workers`));
+      for (let i = 0; i < iterations; i++) {
```

Comment:
This could be optimized to use a worker pool but I'm not convinced the additional complexity justifies the minor performance improvement.